ddsrecorder_participants/include/ddsrecorder_participants/recorder/size_tracker/McapSizeTracker.hpp

```diff
@@ -0,0 +1,197 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file McapSizeTracker.hpp
+ */
+
+#pragma once
+
+#include <filesystem>
+#include <stdexcept>
+
+#include <mcap/mcap.hpp>
+
+#include <cpp_utils/macros/custom_enumeration.hpp>
+
+#include <fastrtps/types/DynamicTypePtr.h>
+
+#include <ddspipe_participants/participant/dynamic_types/ISchemaHandler.hpp>
+
+#include <ddsrecorder_participants/library/library_dll.h>
+
+namespace eprosima {
+namespace ddsrecorder {
+namespace participants {
+
+/**
+ * Class that tracks the MCAP size during recorder execution.
+ *
+ */
+class McapSizeTracker
+{
+public:
+
+    /**
+     * @brief Constructor
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    McapSizeTracker();
+
+    /**
+     * @brief Destructor
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    ~McapSizeTracker();
+
+    /**
+     * @brief Get space needed to write message
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    std::uint64_t get_message_size(
+            const uint64_t& msg_datasize);
+
+    /**
+     * @brief Get space needed to write schema
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    std::uint64_t get_schema_size(
+            const mcap::Schema& schema);
+
+    /**
+     * @brief Get space needed to write blank schema
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    std::uint64_t get_blank_schema_size(
+            const bool& ros2_types,
+            const std::string& schema_name);
+
+    /**
+     * @brief Get space needed to write channel
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    std::uint64_t get_channel_size(
+            const mcap::Channel& channel,
+            const uint64_t& size_metadata);
+
+    /**
+     * @brief Get space needed to write blank channel
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    std::uint64_t get_blank_channel_size(
+            const std::string& channel_name);
+
+    /**
+     * @brief Get space needed to write attachment
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    std::uint64_t get_attachment_size(
+            const std::unique_ptr<fastrtps::rtps::SerializedPayload_t>& serialized_payload);
+
+    /**
+     * @brief Check if there is enough space to write the current file
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    void decrease_mcap_size(
+            const std::uint64_t& size);
+
+    /**
+     * @brief Check if there is enough space to write the current file
+     *
+     */
+    DDSRECORDER_PARTICIPANTS_DllAPI
+    void check_and_increase_mcap_size_final(
```

Comment:
I'd try to find an alternative. If forced to choose a different name, I'd go for something more descriptive.

Reply:
Refactored API.